proposals/299-ip-failure-count.txt

```diff
@@ -191,14 +195,13 @@ Ticket: https://trac.torproject.org/projects/tor/ticket/27491
    packets for a particular IP family while still preventing potential denial
    of service attacks.
 
-8. Pathbias and Failure Probability
+8. Marking Guards As Failed with ClientAutoIPv6ORPort
+
+   If ClientAutoIPv6ORPort is in use, we should mark a guard as failed if we
+   fail both IPv4 and IPv6 connections to the guard.
 
-   If ClientAutoIPv6ORPort is in use, and pathbias is triggered, we should
-   ignore "no route" warnings. The reason for this is because we would be
-   adding two failure points for the failed as described in Section 3 of this
-   proposal. Adding two failure points would make us more likely to prefer the
-   competing IP family over the failed one versus than adding a single failure
-   point on a normal failure.
+   We should add the number of attempts for IPv4 and IPv6 as the failure points
+   of the respective version tried, and halve per Section 5 of this proposal.
```

Comment:
It isn't clear what we are doing here:

I don't understand how this test checks if we are using both IPv4 and IPv6. It seems to contradict the first sentence, which says to check ClientAutoIPv6ORPort.

I also don't understand how we are counting failures separately from "failure points". See my earlier comment about "affinity points".

Reply:
Sorry if it's not mentioned here, but I intended this to mean:

 * The number of attempts would end up as affinity points.

 * We should do it when ClientAutoIPv6ORPort is set using affinity points.

Should we remove this sentence completely (I don't think this section makes too much sense)?

Reply:
I think we should re-write this section based on our conversation above about failed guards.

Reply:
OK.